sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -101,6 +99,15 @@ public EventData(String body) {
         this(Objects.requireNonNull(body, "'body' cannot be null.").getBytes(UTF_8));
     }
 
+    /**
+     * Creates an event with the provided {@link BinaryData} as payload.
+     *
+     * @param body The {@link BinaryData} payload for this event.
+     */
+    public EventData(BinaryData body) {
+        this(Objects.requireNonNull(body, "'body' cannot be null."), new SystemProperties(), Context.NONE);
```

Comment:
The overload this is calling is checking for null. So, it's not required to check here again.